tests/slsa_analyzer/git_service/test_gitlab.py

```diff
@@ -111,3 +114,95 @@ def test_self_hosted_gitlab_without_env_set(tmp_path: Path) -> None:
+    """
+    test_repo_path = Path(__file__).parent.joinpath("resources", "self_hosted_gitlab_repo")
+
+    gitlab_repo = initiate_repo(test_repo_path)
+
+    # Commit untracked test files in the mock repo.
+    # This would only happen the first time this test case is run.
+    if gitlab_repo.repo.untracked_files:
+        commit_files(gitlab_repo, gitlab_repo.repo.untracked_files)
+
+    # Assigning the origin remote url to the passed in parameter.
+    remote_url = request.param
+
+    # For newly init git repositories, the origin remote does not exist. Therefore, we create one and assign it with
+    # ``remote_url``.
+    # If the origin remote already exist we reset its URL to be the ``remote_url``.
+    if "origin" not in gitlab_repo.repo.remotes:
+        remote_name = "origin"
+        gitlab_repo.repo.create_remote(remote_name, remote_url)
+    else:
+        gitlab_repo.repo.remote("origin").set_url(remote_url)
+
+    yield gitlab_repo
+
+    gitlab_repo.clear()
+
+
+# The indirect parameter is used to note that ``self_hosted_gitlab`` should be passed to the ``self_hosted_gitlab``
+# fixture first. The test function would then receive the final fixture value.
+# As for ``expected_origin_url``, this parameter is passed directly into the test function because it's not listed in
+# ``indirect``. Reference:
+# https://docs.pytest.org/en/7.1.x/example/parametrize.html?highlight=indirect#apply-indirect-on-particular-arguments
+@pytest.mark.parametrize(
+    ("self_hosted_gitlab", "expected_origin_url"),
+    [
+        ("https://oauth2:abcxyz@internal.gitlab.org/a/b", "https://internal.gitlab.org/a/b"),
+        ("https://oauth2:abcxyz@internal.gitlab.org/a/b.git", "https://internal.gitlab.org/a/b"),
+    ],
+    indirect=["self_hosted_gitlab"],
+)
+def test_origin_remote_url_masking(self_hosted_gitlab: Git, expected_origin_url: str, tmp_path: Path) -> None:
+    """Test if the ``clone_repo`` and ``check_out_repo`` methods handle masking the token in the clone URL correctly.
+
+    Note that this test ONLY checks if the remote origin URL after the clone/checkout operations is updated back to
+    the token-less URL.
+    It does not check whether those operation works correctly.
+    """
+    user_config_input = """
+    [git_service.gitlab.self_hosted]
+    domain = internal.gitlab.org
+    """
+    user_config_path = os.path.join(tmp_path, "config.ini")
+    with open(user_config_path, "w", encoding="utf-8") as user_config_file:
+        user_config_file.write(user_config_input)
+
+    # We don't have to worry about modifying the ``defaults`` object causing test
+    # pollution here, since we reload the ``defaults`` object before every test with the
+    # ``setup_test`` fixture.
+    load_defaults(user_config_path)
+
+    with mock.patch.dict(os.environ, {"MCN_SELF_HOSTED_GITLAB_TOKEN": "abcxyz"}):
+        gitlab = SelfHostedGitLab()
+        gitlab.load_defaults()
+
```

Comment:
To make sure the token is set and used correctly, can you add a test for 
```
        clone_url = self.construct_clone_url(url)
        repo = git_url.clone_remote_repo(clone_dir, clone_url)
```
and retrieving the remote URL to see if it contains the token?

Reply:
Do you mean add test for each of those method individually? (`construct_clone_url` and `clone_remote_repo`)

Reply:
No I mean to call both in the same test to be able to check if the token is set.

Reply:
Oh okay, I think I get your comment now. The current set of tests only check if the token is not part of the final original remote URL after the operation completes, but I currently don't have any test that check if those operation actually revert to the correct token-included URL for fetching. Is that correct?

Reply:
Right. It's just I currently don't see any ways apart from mocking to capture the actually token-included URL in the call so that we could test if it's using the correct URL :thinking:

Reply:
In additions, there are existing test cases to cover whether this block
```
clone_url = self.construct_clone_url(url)
repo = git_url.clone_remote_repo(clone_dir, clone_url)
```
Would correctly set the token on the URL too.

Reply:
OK, fair enough.